SimpleNotePad.java

```diff
@@ -0,0 +1,198 @@
+import javax.swing.JScrollPane;
+import javax.swing.JTextPane;
+import javax.swing.text.Position;
+import javax.swing.text.StyledDocument;
+
+
+public class SimpleNotePad extends JFrame implements ActionListener{
+    JMenuBar menuBar = new JMenuBar();
+    JMenu fileMenu = new JMenu("File");
+    JMenu editMenu = new JMenu("Edit");
+    JMenu openRecent = new JMenu("Open Recent");
+    JTextPane doc = new JTextPane();
+    List<JMenuItem> fileMenuItems = new ArrayList<>();
+    List<JMenuItem> editMenuItems = new ArrayList<>();
+    List<JMenuItem> openRecentItems = new ArrayList<>();
+
+    OpenRecent openRecentMenu = new OpenRecent(5);
+
+    public SimpleNotePad() {
+        setTitle("A Simple Notepad Tool");
+        //Initialized the MenuItems in two lists based on which menu they will be in.
+        fileMenuItems.add(new JMenuItem("New File"));
+        fileMenuItems.add(new JMenuItem("Save File"));
+        fileMenuItems.add(new JMenuItem("Print File"));
+        fileMenuItems.add(new JMenuItem("Open"));
+
+        editMenuItems.add(new JMenuItem("Copy"));
+        editMenuItems.add(new JMenuItem("Paste"));
+        editMenuItems.add(new JMenuItem("Replace"));
+
+        addToMenu(fileMenuItems, fileMenu);
+        addToMenu(editMenuItems, editMenu);
+
+        //We add the openRecent separately
+        fileMenu.add(openRecent);
+        try {
+            String[] recents = openRecentMenu.getRecents();
+            if(recents.length > 0) {
+                for (String item : recents) {
+                    openRecentItems.add(new JMenuItem(item));
+                }
+            }
+            //I add the open recent items separately because they do not use the default action command
+            for (JMenuItem item : openRecentItems) {
+                openRecent.add(item);
+                item.addActionListener(this);
+                item.setActionCommand("Open Recent" + item.getText());
+            }
+        } catch (BackingStoreException e) {
+        }
+
+        //This sets up the basic frame and adds the menus to the panel.
+        menuBar.add(fileMenu);
+        menuBar.add(editMenu);
+        setJMenuBar(menuBar);
+        add(new JScrollPane(doc));
+        setPreferredSize(new Dimension(600,600));
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setVisible(true);
+        pack();
+    }
+
+    //This method quickly adds every item in the list to the needed menu and gives it a default action command
+    public void addToMenu(List<JMenuItem> list, JMenu menu) {
+        for(JMenuItem item : list) {
+            menu.add(item);
+            item.addActionListener(this);
+            item.setActionCommand(item.getText());
+        }
+    }
+
+    public static void main(String[] args) {
+        SimpleNotePad app = new SimpleNotePad();
+    }
+
+    //This method performs the save file action by opening up a window for the user to choose a save location and name
+    public void save() {

```

Comment:
You could continue decoupling your project by moving all of your functional methods such as save, print, paste, replace, etc into their respective dropdown menu names such as, File.class and Edit.class.